pyprep/utils.py

```diff
@@ -73,28 +75,44 @@ def _mat_quantile(arr, q, axis=None):
     This function mimics MATLAB's logic to produce identical results.
 
     """
-
-    def _mat_quantile_1d(arr, q):
-        arr = arr[~np.isnan(arr)]
-        n = len(arr)
-        if n == 0:
-            return np.NaN
-        elif n == 1:
-            return arr[0]
-        else:
-            q_adj = ((q - 0.5) * n / (n - 1)) + 0.5
-            return np.quantile(arr, np.clip(q_adj, 0, 1))
-
-    # Make sure inputs are both Numpy arrays
-    arr = np.asarray(arr)
-    q = np.asarray(q, dtype=np.float64)
-
-    if axis is not None and len(arr.shape) > 1:
-        # If an axis is specified, calculate quantiles along it
-        return np.apply_along_axis(_mat_quantile_1d, axis, arr, q)
+    # Sort the array in ascending order along the given axis (any NaNs go to the end)
+    # Return NaN if array is empty.
+    arr_sorted = np.sort(arr, axis=axis)
```

Comment:
if axis=None, sort will "flatten" the array. You also flatten the array below again --> if you refactor this to do the sorting AFTER reshaping "...  data into a 2D array with the shape (num_axes, data_per_axis)", then you do one "flatten" less = tiny bit more speed

Reply:
For the array slicing at the bottom to work for both 1D and 2D inputs, 1D arrays need to be shaped into `(1, data_per_axis)` 2D arrays, so the second "flattening" still needs to happen as `sort` produces a 1D array with `axis=None`.

Also FYI I profiled the code again after the fix and ~95% of the time `_mat_quantile` takes now is due to the `sort` function (which is essential and I couldn't figure out any extra speed improvements for) so the total performance impact should be relatively small regardless

Reply:
Ah, makes sense - thanks! Also: if "sort" is the main culprit now, I think we are fine :-)